roles/gpu_operator_bundle_from_commit/files/operator_image_builder_script.yml

```diff
@@ -22,22 +20,37 @@ data:
 
     git show --quiet
 
-    if [ "${BUILDER_FROM_IMAGE}" ]; then
-      # if provided, use custom 'FROM ... as builder' image
-      sed "s|FROM golang:.* as builder|FROM ${BUILDER_FROM_IMAGE} as builder|" -i ${CONTAINER_FILE}
+    CONTAINER_FILE=./Dockerfile
+    if [ -f ${CONTAINER_FILE} ]; then
+      # TODO: Remove this backwards compatible code once we stop testing versions that do not use the new Makefile
+      # This was merged in https://github.com/NVIDIA/gpu-operator/commit/b0e99d9e6ac8738ee335f6e866242b1c9977ea09
+
+      cat <<EOF > /usr/local/bin/docker
+      #! /bin/sh
+      exec podman --cgroup-manager=cgroupfs "\$@"
+      EOF
+      chmod u+x /usr/local/bin/docker
+
+      if [ "${BUILDER_FROM_IMAGE}" ]; then
+        # if provided, use custom 'FROM ... as builder' image
+        sed "s|FROM golang:.* as builder|FROM ${BUILDER_FROM_IMAGE} as builder|" -i ${CONTAINER_FILE}
+      fi
+
+      # avoid docker.io quotas ...
+      sed -i 's|FROM nvidia/cuda:|FROM nvcr.io/nvidia/cuda:|' ${CONTAINER_FILE}
+      make docker-build IMG=${OPERATOR_IMAGE_NAME}
+    else
+      # if provided, use custom 'FROM ... as builder' image by setting BUILDER_IMAGE variable
+      # avoid docker.io quotas by setting CUDA_IMAGE
+      # The docker-image generates an image OUT_IMAGE (based on OPERATOR_IMAGE_NAME)
+      make docker-image \
+        REGISTRY="" \
```

Comment:
The empty `REGISTRY` entry doesn't work. The script should push the image to exactly `OPERATOR_IMAGE_NAME` (and env var passed through the pod definition).

It has a value that looks something like this:
```
image-registry.openshift-image-registry.svc:5000/gpu-operator-ci/gpu-operator-ci:operator_gpu-operator-deadcafe
```

With the new Makefile the empty `REGISTRY` causes the podman command to look something like:

```bash
podman build --pull \
	--tag /gpu-operator:0.0.1-ubi8 \
	--build-arg BASE_DIST="ubi8" \
	--build-arg CUDA_IMAGE="nvcr.io/nvidia/cuda" \
	--build-arg CUDA_VERSION="11.2.1" \
	--build-arg VERSION="0.0.1" \
	--build-arg BUILDER_IMAGE="golang:1.15" \
	--build-arg GOLANG_VERSION="1.15" \
	--file docker/Dockerfile .
```
and the tag `/gpu-operator:0.0.1-ubi8` is invalid.

The makefile should give an option to specify the entire tag rather than just some components of the tag

Reply:
The `podman build` command generates a `/gpu-operator:0.0.1-ubi8` image (I'll update the `REGISTRY`) and this is tagged to match `OUT_IMAGE`. I can add more "smarts" around how the image is generated, but I didn't want to add additional logic to the makefile at this stage -- especially since a lot of our CI logic currently depends on this naming.

If using an intermediate image is a blocker, I can look at reworking the Makefile further.

Reply:
Oh I didn't notice `OUT_IMAGE`. Alright then